MicrosoftStoreServicesSample/Controllers/CollectionsController.cs

```diff
@@ -144,34 +171,69 @@ public async Task<ActionResult<string>> Query([FromBody] ClientCollectionsQueryR
+
+                    if(item.ProductKind == "UnmanagedConsumable")
+                    {
+                        formattedType = "U.Consumable";
+                    }
+
+                    response.AppendFormat("| {0,-12} | {1,-3} | {2,-12} | {3,-11} ",
+                                            item.ProductId,
+                                            quantityToDisplay,
+                                            formattedType,
+                                            item.AcquisitionType);
 
                     //  Check if this is enabled because of a satisfying entitlement from a bundle or subscription
+                    //  format to add those to the output on their own lines.
                     if (item.SatisfiedByProductIds.Any())
                     {
-                        satisfyingEntitlements.Append(" enabled by satisfying entitlement(s) from ");
+                        bool isFirstEntitlement = true;
                         foreach (var parent in item.SatisfiedByProductIds)
                         {
-                            satisfyingEntitlements.Append($"{parent}, ");
+                            if (isFirstEntitlement)
+                            {
+                                isFirstEntitlement = false;
+                                response.AppendFormat("| {0,-12} |\n",
+                                                      parent);
+                            }
+                            else
+                            {
+                                response.AppendFormat("|                                                   {0,-12} |\n",
+                                                      parent);
+                            }
                         }
                     }
-
-                    if (item.ProductKind == "Consumable")
+                    else
                     {
-                        consumableInfo.AppendFormat(" with a balance remaining of {0}", item.Quantity);
+                        response.AppendFormat("|              |\n");
                     }
+                }
 
-                    //  [Product type] [ProductID (aka StoreID from Partner Center)] [Satisfying entitlements] acquired by [acquisition type]
-                    response.AppendFormat("  {0}  {1} {2} acquired by {3}{4}\n",
-                                          item.ProductKind,
-                                          item.ProductId,
-                                          satisfyingEntitlements.ToString(),
-                                          AcquisitionTypeFriendlyName(item.AcquisitionType),
-                                          consumableInfo.ToString());
+                //  If this is from the Client sample, include the JSON so that it can display the items in the UI
+                //  properly
+                if (includeJson)
+                {
+                    response.Append("RawResponse:");

```

Comment:
Should there be a space after the colon? or a newline between them or anything? Formatting of this might look weird.

Reply:
sure, it won't affect the current client parsing the data.